webpack/assets/javascripts/react_app/components/ConfigReports/DiffModal/DiffModal.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Modal, Icon } from 'patternfly-react';
+import PropTypes from 'prop-types';
+
+import DiffContainer from '../../DiffView/DiffContainer';
+import './diffmodal.scss';
+
+class DiffModal extends React.Component {
+  render() {
+    const {
+      log, patch, oldText, newText, isOpen, toggleModal,
+    } = this.props;
+    return (
+      <Modal show={isOpen} onHide={() => toggleModal()} className="diff-modal">
+        <Modal.Header>
+          <h4 id="diff-modal-h4">{log}</h4>
+          <button
+            className="close diff-modal-close"
+            onClick={() => toggleModal()}
+            aria-hidden="true"
+            aria-label="Close"
+          >
+            <Icon type="pf" name="close" />
+          </button>
+        </Modal.Header>
+        <Modal.Body className="diff-modal-body">
+          <DiffContainer patch={patch} oldText={oldText} newText={newText} />
+        </Modal.Body>
+      </Modal>
+    );
+  }
+}
+
+DiffModal.propTypes = {
+  patch: PropTypes.string,
+  oldText: PropTypes.string,
+  newText: PropTypes.string,
+  isOpen: PropTypes.bool.isRequired,
+  toggleModal: PropTypes.func.isRequired,
+};
+
+DiffModal.defaultProps = {
+  patch: '',
+  oldText: '',
+  newText: '',
```

Comment:
should we allow to open it if there is no diff? (Does it even work?)

Reply:
The component won't even render because the parent checks if there is a diff. (Audits and Here).